noir_stdlib/src/merkle/sparse_merkle_tree.nr

```diff
@@ -0,0 +1,84 @@
+use crate::merkle;
+
```

Comment:
Can we add some comments to the functions here and is there a way to test functionality?

Reply:
Sure. Relatedly, I've not noticed any explicit comments/docs on the hash `check_membership` expects the tree to have used. This should probably be added too. I'm guessing it's pedersen?

Reply:
Yeah that would be great!

Reply:
Added

Reply:
Spoke too soon. The test proves, but doesn't verify D:
Guessing this test has exposed a bug that deserves a separate issue?
```
Failed to verify proof
thread 'tests::noir_integration' panicked at 'verification fail for "sparse_merkle_tree"', crates/nargo_cli/tests/prove_and_verify.rs:61:25
```

Reply:
Just noticed when proving from nargo it logs:
```
linear_eval is not 0.
```

Reply:
I think we have an issue with a similar bug, can you link this PR to it as another corpus?

Reply:
Would also be good if you could add a stripped down example into the program (We can revisit this and the other issue once we switch to ultra plonk)